### ISSUE TITLE
Cannot get any components

### ISSUE THREAD

**Issue by bobcroft** [twocolors/node-red-contrib-discovery2mqtt#2]
Hi, I have asked on the Node-RED forum but it seems nobody knows the answer!  I can set up the in and out nodes to my mqtt server and the the green connected flag shows up.  However, no matter what I try I cannot get any components to show.  Node-RED is running under Hassos on the same PC as Home Assistant.  MQTT works perfectly normally.  There are Home assistant topics that start homeassistant/sensor for example.
Can you please advise me what I may be doing wrong.  From the response to my post on the NR forum it seems othewr people are struggling too.

**Comment by twocolors** [comment]
show mqtt topic structure

**Comment by bobcroft** [comment]
This is a topic structure being received in home Assistant
homeassistant/sensor/office/state = {"temperature":"6.24","pressure":"986.73","humidity":81,"rssi":-56}

It is sent from a manually configured mqtt discovery config.

I would be happy to try any topic structure your suggest to prove your nodes are working.

**Comment by twocolors** [comment]
this is not HA discovery structure, more https://www.home-assistant.io/docs/mqtt/discovery/

this is my mqtt (ha discovery structure)

<img width="595" alt="Screenshot 2022-01-05 at 14 31 58" src="https://user-images.githubusercontent.com/207796/148211010-2f59060d-ce96-46ef-b225-d5e2570de423.png">

**Comment by bobcroft** [comment]
Hi, thank you for your reply.

 

The structure I posted was taken from MQTT explorer and it works in HA because I can see the entities in HA and view them in Lovelace.  I am not saying it is right but it does appear to work in HA

From your example I can say that my structure is the same, in MQTT explorer and as far as I can tell, down to the config level.  Please note I don’t doubt your word but I am struggling to understand what aspect of my structure stops your nodes working.

I know it is a PITA helping newbies but if you could elaborate a bit more on what I need to change it would help me greatly.  I have read the document you posted and I didn’t see an issue, but could have missed it.

On the very helpful example you posted some of the expanded ‘sentences’ are truncated on my email screen.

 

Many Thanks

 

Bob

**Comment by bobcroft** [comment]
Hi,

Further to my earlier reply I have copied another working HA discovery config topic, this works in that the entities can be seen in HA configuration and in Lovelace. Is this the correct form?  because I still cannot see any components in your flows.  I have tried on two separate PC’s and two independent HA set ups.

192.168.1.196/homeassistant/sensor/ volcano_taranakiegmont_activity

config = {"name":"Taranaki/Egmont Activity Description","state_topic":"homeassistant/sensor/volcano_taranakiegmont/state","value_template":"{{ value_json.activity }}"}

 

}}"}

**Comment by twocolors** [comment]
yes, that is the correct format. but only sernsors and switches are supported.

if you have esphome or tasmota enable discovery and see format of use
